omou-frontend/src/components/FeatureViews/Registration/Form.js

```diff
@@ -150,31 +142,45 @@ class Form extends Component {
     handleNext() {
         this.setState((oldState) => {
             if (this.validateSection()) {
-                const conditionalField = this.getConditionalFieldFromCurrentSection();
+                const conditionalField = this.getConditionalFieldFromCurrentSection(),
+                    nextActiveStep = oldState.activeStep + 1,
+                    nextActiveSection = this.getFormObject().section_titles[nextActiveStep];
                 let newState = {
-                    activeStep: oldState.activeStep + 1,
-                    activeSection: this.getFormObject().section_titles[oldState.activeStep + 1],
+                    activeStep: nextActiveStep,
+                    activeSection: nextActiveSection,
                     conditional: conditionalField ? conditionalField : oldState.conditional,
                     nextSection: true,
                 };
                 if (conditionalField) {
                     let formContents = this.getFormObject(),
-                        title = this.getFormObject().section_titles[oldState.activeStep + 1];
+                        title = nextActiveSection;
                     // create blank fields based on form type
                     newState[title] = {};
-                    formContents[this.getFormObject().section_titles[oldState.activeStep + 1]][conditionalField].forEach((field)=>{
+                    formContents[nextActiveSection][conditionalField].forEach((field)=>{
                         newState[title][field.field] = undefined;
                     });
                     // create validated state for each field
                     newState[`${title}_validated`] = {};
-                    formContents[this.getFormObject().section_titles[oldState.activeStep + 1]][conditionalField].forEach((field)=>{
+                    formContents[nextActiveSection][conditionalField].forEach((field)=>{
                         newState[`${title}_validated`][field.field] = true;
                     });
                 }
                 return newState;
             } else {
                 return {};
             }
+        }, () => {
+            const field = this.getActiveSection().find((matchingField) => matchingField.type === "course");
+            if (field) {
+                let course = decodeURIComponent(this.props.match.params.course);
+                course = this.props.courses.find((matchingCourse) => course === matchingCourse.course_title);
+                // convert it to a format that onselectChange can use
+                course = {
```

Comment:
The logic of when to setting the value for courses shouldn't be in `handleNext()` in the case that the course is the first section (which is highly likely - we want to put course selection as the first section)